From 6d2c8406f648c5eddc80803d849b7acc675b8294 Mon Sep 17 00:00:00 2001
From: Dilek Akcay <dilek.akcay@qt.io>
Date: Fri, 26 Sep 2025 18:27:21 +0200
Subject: [PATCH 082/239] SearchField: Update SortFilterProxyModel in docs

The documentation snippet used QSortFilterProxyModel, while the Gallery
example already demonstrates the QML SortFilterProxyModel. Update the doc
to use QML SFPM for consistency.

Change-Id: I65423d2a36625e5b6a56c734808d8ea735d55b02
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 939fadbbf43fc3d70881255ca015eed30f06fb6d)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quicktemplates/qquicksearchfield.cpp | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/src/quicktemplates/qquicksearchfield.cpp b/src/quicktemplates/qquicksearchfield.cpp
index b8162f8152..feddc1208f 100644
--- a/src/quicktemplates/qquicksearchfield.cpp
+++ b/src/quicktemplates/qquicksearchfield.cpp
@@ -57,11 +57,24 @@ QT_BEGIN_NAMESPACE
         ListElement { name: "WaterMelon"; color: "pink" }
     }
 
-    QSortFilterProxyModel {
+    SortFilterProxyModel {
         id: fruitFilter
-        sourceModel: fruitModel
-        filterRegularExpression: RegExp(fruitSearch.text, "i")
-        filterRole: 0 // needs to be set explicitly
+        model: fruitModel
+        sorters: [
+            RoleSorter {
+                roleName: "name"
+            }
+        ]
+        filters: [
+            FunctionFilter {
+                component CustomData: QtObject { property string name }
+                property var regExp: new RegExp(fruitSearch.text, "i")
+                onRegExpChanged: invalidate()
+                function filter(data: CustomData): bool {
+                    return regExp.test(data.name);
+                }
+            }
+        ]
     }
 
     SearchField {
-- 
2.51.2

